lib/src/git.rs

```diff
@@ -2034,24 +2034,24 @@ where
     // detect whether one is present with the current API, and `jj git remote
     // set-url` refuses to work with them anyway.
     new_remote = new_remote.with_fetch_tags(remote.fetch_tags());
-    for direction in [gix::remote::Direction::Fetch, gix::remote::Direction::Push] {
-        new_remote
-            .replace_refspecs(
-                remote
-                    .refspecs(direction)
-                    .iter()
-                    .map(|refspec| refspec.to_ref().to_bstring()),
-                direction,
-            )
-            .expect("existing refspecs to be valid");
-    }
+    new_remote
```

Comment:
Is it okay to drop push refspecs?

Reply:
Thank you for spotting this! I actually misunderstood the whole point of this function. It's now reworked.